### PR TITLE
feat: add get properties of archived workflow API (#477)

### LINKED ISSUE #477: Get the properties of an archived workflow API
Provide a workflow API method to get the properties of an archived z/OSMF workflow on a z/OS system.

The implementation should support the z/OSMF **Get the properties of an archived Workflow** REST API and expose the required request/response handling through the SDK.

IBM REST API reference:
https://www.ibm.com/docs/en/zos/3.2.0?topic=services-get-properties-archived-workflow

### PR DESCRIPTION
**Pull request by Eshaan-byte**
## Summary
- Add `getArchivedProperties` methods to `WorkflowGet` to support the z/OSMF 
  GET `/archivedworkflows/{workflowKey}` REST API (issue #477)
- Support optional `returnData` query parameter for requesting step and variable information
- Reuse existing `WorkflowGetPropertiesInputData` and `WorkflowGetPropertiesResponse` classes 
  since the archived endpoint shares the same request parameters and response structure
- Add unit tests for URL generation, response parsing, and null validation

## API Reference
- [IBM z/OSMF: Get the properties of an archived Workflow](https://www.ibm.com/docs/en/zos/3.2.0?topic=services-get-properties-archived-workflow)

## Test plan
- [x] `tstWorkflowGetArchivedPropertiesByKeySuccess` - verifies URL and response parsing
- [x] `tstWorkflowGetArchivedPropertiesWithReturnDataUrlGeneration` - verifies `?returnData=steps,variables`
- [x] `tstWorkflowGetArchivedPropertiesCommonReturnStepsOnlyUrlGeneration` - verifies `?returnData=steps`
- [x] `tstWorkflowGetArchivedPropertiesWithNullInputData` - null input validation
- [x] `tstWorkflowGetArchivedPropertiesWithNullWorkflowKey` - null key validation

Closes #477

### CODE REVIEW

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java [hunk 1]
```diff
@@ -222,12 +218,86 @@ public WorkflowGetPropertiesResponse getPropertiesCommon(final WorkflowGetProper
...
+
+    /**
+     * Get the properties of an archived z/OSMF workflow.
+     *
+     * @param propertiesInputData workflow properties retrieval parameters
+     * @return archived workflow properties returned by z/OSMF
+     * @throws ZosmfRequestException request error state
+     */
+    public WorkflowGetPropertiesResponse getArchivedPropertiesCommon(
+            final WorkflowGetPropertiesInputData propertiesInputData) throws ZosmfRequestException {
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java]: This introduces unnecessary duplication. The getArchivedPropertiesCommon() method is redundant and increases maintenance overhead.

We should refactor this to reuse getPropertiesCommon() instead, and introduce a boolean parameter (e.g., isArchived) to control which URL is used internally.

This keeps the logic centralized, reduces code duplication, and makes future changes easier since URL handling would only need to be updated in one place.

**Review by frankgiordano — CHANGES_REQUESTED** [review]

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java (hunk 1 above)
Comment (Eshaan-byte) [reply src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java]: Thanks for the feedback @frankgiordano. I've refactored as suggested, getArchivedPropertiesCommon() now delegates to a private getPropertiesCommon(propertiesInputData, isArchived) method that both getPropertiesCommon() and getArchivedPropertiesCommon() share. The isArchived boolean controls whether ARCHIVED_WORKFLOWS_RESOURCE or WORKFLOWS_RESOURCE is used for the URL. All logic is centralised in one place now.

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java [hunk 2]
```diff
@@ -88,12 +88,11 @@ public WorkflowGetDefinitionResponse getDefinition(final String definitionFilePa
...
-     *                                     path name (including the file name) or a fully qualified z/OS data set name
+     * @param definitionFilePath specifies the location of the workflow definition file, which is either a UNIX
+     *                           path name (including the file name) or a fully qualified z/OS data set name
      * @param workflowDefinitionFileSystem nickname of the system on which the workflow definition file resides
      * @return workflow definition details returned by z/OSMF
      * @throws ZosmfRequestException request error state
-     * @author Ashish Kumar Dash
      */
     public WorkflowGetDefinitionResponse getDefinition(final String definitionFilePath,
                                                        final String workflowDefinitionFileSystem)
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java:102]: remove this method it is not needed. see discussion in discord.

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java [hunk 3]
```diff
@@ -112,7 +111,6 @@ public WorkflowGetDefinitionResponse getDefinition(final String definitionFilePa
      * @param definitionInputData workflow definition retrieval parameters
      * @return workflow definition details returned by z/OSMF
      * @throws ZosmfRequestException request error state
-     * @author Ashish Kumar Dash
      */
     public WorkflowGetDefinitionResponse getDefinitionCommon(final WorkflowGetDefinitionInputData definitionInputData)
             throws ZosmfRequestException {
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java:121]: rename this method to getDefinitionByInputData

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java [hunk 4]
```diff
@@ -167,7 +164,6 @@ public WorkflowGetPropertiesResponse getProperties(final String workflowKey) thr
      * @param returnVariables whether the response includes the workflow variable information
      * @return workflow properties returned by z/OSMF
      * @throws ZosmfRequestException request error state
-     * @author Ashish Kumar Dash
      */
     public WorkflowGetPropertiesResponse getProperties(final String workflowKey,
                                                        final boolean returnSteps,
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java]: remove this method.

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java (hunk 2 above)
Comment (frankgiordano) [reply src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java:102]: never mind wrong method

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java (hunk 3 above)
Comment (frankgiordano) [reply src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java:121]: never mind

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java (hunk 4 above)
Comment (frankgiordano) [reply src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java]: see discord discussion.

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java [hunk 5]
```diff
@@ -187,14 +183,74 @@ public WorkflowGetPropertiesResponse getProperties(final String workflowKey,
...
+     * @param workflowKey     workflow key that uniquely identifies the archived workflow instance
+     * @param returnSteps     whether the response includes the workflow step information
+     * @param returnVariables whether the response includes the workflow variable information
+     * @return archived workflow properties returned by z/OSMF
+     * @throws ZosmfRequestException request error state
+     */
+    public WorkflowGetPropertiesResponse getArchivedProperties(final String workflowKey,
+                                                                final boolean returnSteps,
+                                                                final boolean returnVariables)
+            throws ZosmfRequestException {
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java]: remove this method it is not needed see discord discussion.

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java (hunk 1 above)
Comment (frankgiordano) [reply src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java]: remove public WorkflowGetPropertiesResponse getArchivedPropertiesCommon(
            final WorkflowGetPropertiesInputData propertiesInputData) throws ZosmfRequestException {

and use 

private WorkflowGetPropertiesResponse getPropertiesCommon(
            final WorkflowGetPropertiesInputData propertiesInputData,
            final boolean isArchived) throws ZosmfRequestException {

only

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java [hunk 6]
```diff
@@ -187,14 +183,74 @@ public WorkflowGetPropertiesResponse getProperties(final String workflowKey,
...
+
+    /**
+     * Internal method that retrieves workflow properties from either the workflows or archived workflows endpoint.
+     *
+     * @param propertiesInputData workflow properties retrieval parameters
+     * @param isArchived          whether to use the archived workflows endpoint
+     * @return workflow properties returned by z/OSMF
+     * @throws ZosmfRequestException request error state
+     */
+    private WorkflowGetPropertiesResponse getPropertiesCommon(
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java]: make this public method.

**Review by frankgiordano — APPROVED** [review]

File: src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java [hunk 7]
```diff
@@ -187,14 +183,74 @@ public WorkflowGetPropertiesResponse getProperties(final String workflowKey,
      * @param propertiesInputData workflow properties retrieval parameters
      * @return workflow properties returned by z/OSMF
      * @throws ZosmfRequestException request error state
-     * @author Ashish Kumar Dash
      */
     public WorkflowGetPropertiesResponse getPropertiesCommon(final WorkflowGetPropertiesInputData propertiesInputData)
             throws ZosmfRequestException {
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosmfworkflow/methods/WorkflowGet.java]: remove this method.

**Review by frankgiordano — CHANGES_REQUESTED** [review]
